mem/buffer_pool.go

```diff
@@ -109,6 +111,132 @@ func (p *tieredBufferPool) getPool(size int) BufferPool {
 	return p.sizedPools[poolIdx]
 }
 
+type binaryTieredBufferPool struct {
+	indexOfNextLargestBit     []int
+	indexOfPreviousLargestBit []int
```

Comment:
It is not very intuitive what these slices represent. Could you please add some comments?